### ISSUE TITLE
Build issues on on M3 Mac

### ISSUE THREAD

**Issue by jpziegler** [cvc5/cvc5#10973]
**Describe the bug**

On my M3 Mac, when building with:
```
./configure.sh --python-bindings --auto-download --poly --cocoa
cd build
make
```
I get the following error:
```
[  3%] Performing configure step for 'CoCoA-EP'
CMake Error at /Users/user/projects/cvc5/build/deps/src/CoCoA-EP-stamp/CoCoA-EP-configure-Production.cmake:37 (message):
  Command failed: 1

   './configure' '--prefix=/Users/user/projects/cvc5/build/deps' '--with-libgmp=/opt/homebrew/lib/libgmp.dylib' '--with-cxx=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++'

  See also

    /Users/user/projects/cvc5/build/deps/src/CoCoA-EP-stamp/CoCoA-EP-configure.log


-- Log output is:
Starting configuration process for CoCoALib...

ERROR: Are you sure "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++" is a C++ compiler?   [[verify-compiler.sh]]
./configure: ERROR: Check the options --with-cxx=... and --with-cxxflags=...
./configure: ERROR: (or the environment variables CXX and CXXFLAGS).

CMake Error at /Users/user/projects/cvc5/build/deps/src/CoCoA-EP-stamp/CoCoA-EP-configure-Production.cmake:47 (message):
  Stopping after outputting logs.


make[2]: *** [deps/src/CoCoA-EP-stamp/CoCoA-EP-configure] Error 1
make[1]: *** [CMakeFiles/CoCoA-EP.dir/all] Error 2
make: *** [all] Error 2
```
**cvc5 version/commit**: 
```
% git log -1
commit 99bfe0bcc00bf730c84db499b7e27419bf165dc3 (HEAD -> main, origin/main, origin/HEAD)
```
**Operating system**: Name/version of the operating system.  
macOS Sonoma 14.5

**`configure.sh` options**  
```--python-bindings --auto-download --poly --cocoa```

**`configure.sh` output**  
```
% ./configure.sh
-- The C compiler identification is AppleClang 15.0.0.15000309
-- The CXX compiler identification is AppleClang 15.0.0.15000309
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: /usr/bin/git (found version "2.39.3 (Apple Git-146)")
-- Building Production build
-- Performing Test HAVE_C_FLAG_O3
-- Performing Test HAVE_C_FLAG_O3 - Success
-- Configuring with C flag '-O3'
-- Performing Test HAVE_CXX_FLAG_O3
-- Performing Test HAVE_CXX_FLAG_O3 - Success
-- Configuring with CXX flag '-O3'
-- Performing Test HAVE_C_FLAG_Wall
-- Performing Test HAVE_C_FLAG_Wall - Success
-- Configuring with C flag '-Wall'
-- Performing Test HAVE_CXX_FLAG_Wall
-- Performing Test HAVE_CXX_FLAG_Wall - Success
-- Configuring with CXX flag '-Wall'
-- Performing Test HAVE_C_FLAG_Wunused_private_field
-- Performing Test HAVE_C_FLAG_Wunused_private_field - Success
-- Configuring with C flag '-Wno-unused-private-field'
-- Performing Test HAVE_CXX_FLAG_Wunused_private_field
-- Performing Test HAVE_CXX_FLAG_Wunused_private_field - Success
-- Configuring with CXX flag '-Wno-unused-private-field'
-- Performing Test HAVE_C_FLAG_Wdangling_reference
-- Performing Test HAVE_C_FLAG_Wdangling_reference - Failed
-- Performing Test HAVE_CXX_FLAG_Wdangling_reference
-- Performing Test HAVE_CXX_FLAG_Wdangling_reference - Failed
-- Performing Test HAVE_C_FLAG_fexceptions
-- Performing Test HAVE_C_FLAG_fexceptions - Success
-- Configuring with C flag '-fexceptions'
-- Performing Test HAVE_CXX_FLAG_Wsuggest_override
-- Performing Test HAVE_CXX_FLAG_Wsuggest_override - Success
-- Configuring with CXX flag '-Wsuggest-override'
-- Performing Test HAVE_CXX_FLAG_Wnon_virtual_dtor
-- Performing Test HAVE_CXX_FLAG_Wnon_virtual_dtor - Success
-- Configuring with CXX flag '-Wnon-virtual-dtor'
-- Performing Test HAVE_C_FLAG_Wimplicit_fallthrough
-- Performing Test HAVE_C_FLAG_Wimplicit_fallthrough - Success
-- Configuring with C flag '-Wimplicit-fallthrough'
-- Performing Test HAVE_CXX_FLAG_Wimplicit_fallthrough
-- Performing Test HAVE_CXX_FLAG_Wimplicit_fallthrough - Success
-- Configuring with CXX flag '-Wimplicit-fallthrough'
-- Performing Test HAVE_C_FLAG_Wshadow
-- Performing Test HAVE_C_FLAG_Wshadow - Success
-- Configuring with C flag '-Wshadow'
-- Performing Test HAVE_CXX_FLAG_Wshadow
-- Performing Test HAVE_CXX_FLAG_Wshadow - Success
-- Configuring with CXX flag '-Wshadow'
-- Performing Test HAVE_C_FLAG_fno_operator_names
-- Performing Test HAVE_C_FLAG_fno_operator_names - Success
-- Configuring with C flag '-fno-operator-names'
-- Performing Test HAVE_CXX_FLAG_fno_operator_names
-- Performing Test HAVE_CXX_FLAG_fno_operator_names - Success
-- Configuring with CXX flag '-fno-operator-names'
-- Performing Test HAVE_CXX_FLAG_fno_extern_tls_init
-- Performing Test HAVE_CXX_FLAG_fno_extern_tls_init - Failed
-- Performing Test HAVE_CXX_FLAG_Wclass_memaccess
-- Performing Test HAVE_CXX_FLAG_Wclass_memaccess - Failed
-- Disabling unit tests since assertions are disabled.
-- Found Python: /Users/user/.pyenv/shims/python3 (found version "3.12.2") found components: Interpreter
-- Found GMP (unknown version): /opt/homebrew/lib/libgmp.dylib
-- Looking for getc_unlocked
-- Looking for getc_unlocked - not found
-- Building CaDiCaL rel-1.7.4: /Users/user/projects/cvc5/build/deps/lib/libcadical.a
-- Building Poly 0.1.13: /Users/user/projects/cvc5/build/deps/lib/libpoly.dylib, /Users/user/projects/cvc5/build/deps/lib/libpolyxx.dylib
-- Building SymFPU: /Users/user/projects/cvc5/build/deps/include/
-- Performing Test CVC5_NEED_INT64_T_OVERLOADS
-- Performing Test CVC5_NEED_INT64_T_OVERLOADS - Success
-- Performing Test CVC5_NEED_HASH_UINT64_T_OVERLOAD
-- Performing Test CVC5_NEED_HASH_UINT64_T_OVERLOAD - Failed
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for sys/wait.h
-- Looking for sys/wait.h - found
-- Looking for C++ include ext/stdio_filebuf.h
-- Looking for C++ include ext/stdio_filebuf.h - not found
-- Looking for clock_gettime
-- Looking for clock_gettime - found
-- Looking for ffs
-- Looking for ffs - found
-- Looking for optreset
-- Looking for optreset - found
-- Looking for sigaltstack
-- Looking for sigaltstack - found
-- Looking for strerror_r
-- Looking for strerror_r - found
-- Looking for strtok_r
-- Looking for strtok_r - found
-- Looking for setitimer
-- Looking for setitimer - found
-- Performing Test STRERROR_R_CHAR_P
-- Performing Test STRERROR_R_CHAR_P - Failed
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success

cvc5 1.1.3-dev

Build profile             : production

Assertions                : off
Debug symbols             : off
Debug context mem mgr     : off

Muzzle                    : off
Statistics                : on
Tracing                   : off

ASan                      : off
UBSan                     : off
TSan                      : off
Coverage (gcov)           : off
Profiling (gprof)         : off
Unit tests                : off
Valgrind                  : off

Shared build              : on
Python bindings           : off
Java bindings             : off
Interprocedural opt.      : off

CryptoMiniSat             : off
GLPK                      : off
Kissat                    : off
LibPoly                   : on (local)
CoCoALib                  : off
MP library                : gmp (system)
Editline                  : off

Api docs                  : off


CPPLAGS (-D...): NDEBUG CVC5_STATISTICS_ON CVC5_USE_POLY
CXXFLAGS       : -O3 -Wall -Wno-unused-private-field -Wsuggest-override -Wnon-virtual-dtor -Wimplicit-fallthrough -Wshadow -fno-operator-names
CFLAGS         : -O3 -Wall -Wno-unused-private-field -fexceptions -Wimplicit-fallthrough -Wshadow -fno-operator-names
Linker flags   : 

Install prefix : /usr/local

cvc5 license: modified BSD

Note that this configuration is NOT built against any GPL'ed libraries, so
it is covered by the (modified) BSD license.  To build against GPL'ed
libraries which can improve cvc5's performance on arithmetic and bit-vector
logics, use the 'configure.sh' script to re-configure with '--best --gpl'.

Now change to 'build' and type 'make', followed by 'make check' or 'make install'.

-- Configuring done (4.2s)
-- Generating done (0.3s)
```

**Comment by jpziegler** [comment]
Manually changing line 4 of the generated file `build/deps/src/CoCoA-EP-stamp/CoCoA-EP-configure-Production.cmake` as follows allows the build to succeed.
```
set(command "./configure;--prefix=/Users/jared.ziegler/projects/cvc5/build/deps;--with-libgmp=/opt/homebrew/lib/libgmp.dylib;--with-cxx=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++")
```
```
set(command "./configure;--prefix=/Users/jared.ziegler/projects/cvc5/build/deps;--with-libgmp=/opt/homebrew/lib/libgmp.dylib")
```

**Comment by alex-ozdemir** [comment]
Thank you for this report, @jpziegler.
We'd like to this fix this, without reverting the issue fixed by #10463.

Two questions for you:
* Does `/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++` exist on your system, and is it a C++ compiler?
* When you make the change that you describe, which C++ compiler gets used by CoCoA? (Perhaps you could send us the output of `cat build/deps/src/CoCoA-EP-stamp/*.log`?)
